frontend/ui/uimanager.lua

```diff
@@ -710,19 +701,16 @@ function UIManager:getSecondTopmostWidget()
     -- Because everything is terrible, you can actually instantiate multiple VirtualKeyboards,
     -- and they'll stack at the top, so, loop until we get something that *isn't* VK...
     for i = #self._window_stack - 1, 1, -1 do
-        local sec = self._window_stack[i]
-        if not sec or not sec.widget then
-            return nil
-        end
+        local widget = self._window_stack[i].widget
 
-        if sec.widget.name then
-            if sec.widget.name ~= "VirtualKeyboard" then
-                return sec.widget.name
+        if widget.name then
+            if widget.name ~= "VirtualKeyboard" then
+                return widget.name
```

Comment:
Curious if the `name` property is special that itself is a widget or this might be a bug?

Reply:
I'm... not sure I understand the question?

(Very few widgets have the `name` field set, ReaderUI, FileManager & VirtualKeyboard are the main contenders).

Reply:
Oh, yeah, no, that's on purpose ;). If it returns a string it's the name, if it returns a ref it's a ref to the widget, it return nil if no match ;).

The intended use-case was comparing against a *known* widget, so an equality test against a string or a ref doesn't change a thing for the caller.

I can see that you and autodim use it to actually poke at the widget's internal fields, in which case, you should definitely type-check the result (or tweak it to return a name, widget tuple if you find it more convenient). ;).

Reply:
If you need something to test, the frontlight widget is named, too, and I think it popping up above the vocabbuilder might be entirely pluasible?

Reply:
Thanks for the explanation :) I use it only when a wiki window is shown and I try to check if it’s from a dictionary window so to inherit the latter’s button tweaks. For now I don’t see it possible to have a wiki window above a nil widget, so it should be fine, I suppose :) (just checked strings can be dotted too to get a nil so won’t break the code)

Reply:
Oh, right, every string gets a metatable that points to the string standard library, so field access will work (funky!).